benchmark.go

```diff
@@ -188,12 +213,57 @@ func runBenchmark(out output, logger *zap.Logger, allOpts Options, resolved reso
 		zap.Time("startTime", start),
 	)
 
+    if opts.Format == "json" || opts.Format == "JSON" {
+        // Calling JSON output helper method
+        outputJSON(opts, overall, out, goMaxProcs, numConns, total)
+    } else {
+        // Calling plaintext output helper method
+        outputPlaintext(opts, overall, out, goMaxProcs, numConns, total)
+	}
+}
+
+// JSON output helper method
+func outputJSON(opts BenchmarkOptions, overall *benchmarkState, out output, goMaxProcs int, numConns int, total time.Duration) {
```

Comment:
generally for readability, consider adding extra blank lines to separate logic inside a function. E.g. take a look a `getQuantile` function on how it does that.